### PR TITLE
Yaml parsing refactor

### LINKED ISSUE #2801: h121 pps1 pin conflict
in hellen121vag yaml looks like same EFI_ADC_3 was used twice: PPF1 and IN_RES2

1) short term need a fix in hellen121 yaml
2) long term open question if pinout generator should be failing faster in case of similar pin conflict?

### PR DESCRIPTION
**Pull request by chuckwagoncomputing**
This fixes #2801 and #2810

### CODE REVIEW

File: java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java [hunk 1]
```diff
@@ -414,75 +405,126 @@ private static void processYamlFile(File yamlFile, ReaderState state,
...
                 }
-            } else if (idObject instanceof String ) {
-                findMatchingEnum((String) idObject, (String) pin.get("ts_name"), (String) pin.get("class"), state, listOutputs, listAnalogInputs, listEventInputs, listSwitchInputs);
+            } else if (pinId instanceof String ) {
+                Map<String, Object> thisPin = new HashMap();
+                thisPin.put("id", pinId);
+                thisPin.put("ts_name", pinName);
+                thisPin.put("class", pinClass);
+                listPins.add(thisPin);
             }
```
Comment (mck1117) [inline java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java:432]: we should throw in the `else { }` case if it's neither an array nor string

File: java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java [hunk 2]
```diff
@@ -414,75 +405,126 @@ private static void processYamlFile(File yamlFile, ReaderState state,
...
-                list.put(kv.getValue().getIntValue(), ts_name);
-            } else if (kv.getKey().equals(nothingName)) {
-                list.put(kv.getValue().getIntValue(), "NONE");
+        for (Map.Entry<String, ArrayList<String>> kv : names.entrySet()) {
+            String outputEnumName = "";
+            String pinType = "";
+            String nothingName = "";
+            switch (kv.getKey()) {
+                case "outputs":
+                    outputEnumName = "output_pin_e_enum";
```
Comment (rusefillc) [inline java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java:486]: I've just pushed https://github.com/rusefi/rusefi/blob/master/java_tools/configuration_definition/src/com/rusefi/PinType.java

please refactor that enum to your liking

open question if find() method should throw exception or return null for unexpected? at the moment I am not sure why do we not handle "else" as a failure

File: java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java (hunk 1 above)
Comment (chuckwagoncomputing) [reply java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java:432]: I should be able to get the class name with pinId.getClass().getCanonicalName(), right? Or do I want getSimpleName()?

Comment (chuckwagoncomputing) [reply java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java:432]: Oh or getName()

Comment (mck1117) [reply java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java:434]: something like that, yeah

File: java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java (hunk 2 above)
Comment (chuckwagoncomputing) [reply java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java:488]: oh nice. I will use this. tomorrow.

**Comment by chuckwagoncomputing** [comment]
I think this is ready to go. I have verified that the output matches.

**Pull request merged**